### PR TITLE
Introduce an option to suppress metadata sections generated by `moc`

### PR DESCRIPTION
**Pull request by ggreif**
... called `--omit-metadata`, with usage is analogous to `--public-metadata`.

This is mostly for privacy or CI purposes, thus no tests added here, but will be exercised by CI in a later PR.

See e.g. #3163.

Contains some minor refactoring.

### CODE REVIEW

File: src/exes/moc.ml [hunk 1]
```diff
@@ -89,6 +89,12 @@ let argspec = [
       String.concat " or " valid_metadata_names ^
       ") as `public` (default is `private`)";
 
+  "--suppress-metadata",
+    Arg.String (fun n -> Flags.(suppress_metadata_names := n :: !suppress_metadata_names)),
```
Comment (ggreif) [inline src/exes/moc.ml:93]: Would it make sense to verify that `n` is in `valid_metadata_names`?

Comment (nomeata) [reply src/exes/moc.ml:93]: This is one of those questions that nobody will ever say no to :-)

File: Changelog.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
 # Motoko compiler changelog
 
+## FUTURE (2022-03-XX)
+
+* motoko (`moc`)
+
+  * Adds new flag `--suppress-metadata` to omit certain metadata sections from `actor` (and `actor class`) Wasm (#3164)
```
Comment (crusso) [inline Changelog.md:7]: hmm, maybe --omit-metadata is actually a better name for this?

File: src/exes/moc.ml (hunk 1 above)
Comment (crusso) [reply src/exes/moc.ml:93]: Yes. And Yes,

Comment (ggreif) [reply src/exes/moc.ml:93]: there is `process_public_metadata_names`  already...

File: Changelog.md (hunk 2 above)
Comment (ggreif) [reply Changelog.md:7]: Yeah, why not?

Comment (crusso) [reply Changelog.md:7]: Could you add a line to the compiler options adoc file too?

**Review by crusso — APPROVED** [review]
LGTM, modulo comments. Thanks for tidying up too.

File: src/exes/moc.ml (hunk 1 above)
Comment (ggreif) [reply src/exes/moc.ml:93]: Done:  ee533c6

File: Changelog.md (hunk 2 above)
Comment (ggreif) [reply Changelog.md]: Done in 864326878c76c3ad09510f6370edbe5818a1ba1b and ed9cbcf03.

Comment (ggreif) [reply Changelog.md]: And 47e57d527. PTAL.

**Review by crusso — APPROVED** [review]

**Pull request merged**